instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/item/ItemTracer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.item;
+
+import static io.opentelemetry.api.trace.Span.Kind.INTERNAL;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import java.lang.ref.Reference;
+import java.lang.ref.WeakReference;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.springframework.batch.core.scope.context.ChunkContext;
+
+public class ItemTracer extends BaseTracer {
+  private static final ItemTracer TRACER = new ItemTracer();
+
+  public static ItemTracer tracer() {
+    return TRACER;
+  }
+
+  // item-level listeners do not receive chunk/step context as parameters
+  // fortunately the whole chunk always executes on one thread - in Spring Batch chunk is almost
+  // synonymous with a DB transaction
+  // that makes it quite safe to store the current chunk context in a thread local and use it in
+  // each item span
+
+  private final ThreadLocal<WeakReference<ChunkContext>> currentChunk = new ThreadLocal<>();
+
+  public void startChunk(ChunkContext chunkContext) {
+    currentChunk.set(new WeakReference<>(chunkContext));
+  }
+
+  public void endChunk() {
+    currentChunk.remove();
+  }
+
+  @Nullable
+  public Context startReadSpan() {
+    return startItemSpan("ItemRead");
+  }
+
+  @Nullable
+  public Context startProcessSpan() {
+    return startItemSpan("ItemProcess");
+  }
+
+  @Nullable
+  public Context startWriteSpan() {
+    return startItemSpan("ItemWrite");
+  }
+
+  @Nullable
+  private Context startItemSpan(String itemOperationName) {
+    ChunkContext chunkContext =
+        Optional.ofNullable(currentChunk.get()).map(Reference::get).orElse(null);
+    if (chunkContext == null) {
+      return null;
+    }
+
+    String jobName = chunkContext.getStepContext().getJobName();
+    String stepName = chunkContext.getStepContext().getStepName();
+    Span span =
+        tracer
+            .spanBuilder("BatchJob " + jobName + "." + stepName + "." + itemOperationName)
```

Comment:
It'd definitely be nice to get some spec issues going for this batch processing stuff

Reply:
Yes, definitely - I've filed a [spec issue](https://github.com/open-telemetry/opentelemetry-specification/issues/1347) about that (now that my vacation is over I can start working on that 😄 ).